### PR TITLE
task: add integration tests for tests-scan

### PR DESCRIPTION
**Pull request by jelly**
* [x] rebase
* [x] factorize mock server
* [x] assert complete generated command,
* [x] add test for human mode 
* [x] test zero exit code
* [x] test missing PR with program failure

### CODE REVIEW

File: task/test-tests-scan [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        self.assertIsNone(stderr)
+
+    def test_no_arguments_human_readable(self):
+        args = ["--dry", "--repo", self.repo, "--context", self.context, "-v"]
+        proc, output, stderr = self.run_tests_scan(args)
+
+        self.assertEqual(proc.returncode, 0)
+        expected_output = self.expected_human_output()
+        self.assertEqual(output.strip(), expected_output)
+        self.assertIsNone(stderr)
```
Comment (jelly) [inline task/test-tests-scan:175]: nitpick, `ifmain` is missinghere.

File: task/test-tests-scan [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+        self.context = "fedora/nightly"
+        self.revision = "abcdef"
+
+    def tearDown(self):
+        self.server.kill()
+        shutil.rmtree(self.temp)
+
+    def run_tests_scan(self, args):
+        script = os.path.join(BOTS_DIR, "tests-scan")
+        print([script, *args])
```
Comment (martinpitt) [inline task/test-tests-scan:89]: debugging leftover

File: task/test-tests-scan [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+        return proc, output, stderr
+
+    def expected_command(self):
+        expected_output = f"""./s3-streamer --repo {self.repo} --test-name pull-{self.pull_number}-\\d+-\\d+
+--github-context {self.context} --revision {self.revision} -- /bin/sh -c "PRIORITY=0005 ./make-checkout --verbose
+--repo={self.repo} {self.revision} && cd make-checkout-workdir && TEST_OS=fedora BASE_BRANCH={self.revision}
+COCKPIT_BOTS_REF=main TEST_SCENARIO=nightly ../tests-invoke --pull-number {self.pull_number}
+--revision {self.revision} --repo {self.repo}"
+"""
+        return " ".join(expected_output.splitlines()).strip()
```
Comment (martinpitt) [inline task/test-tests-scan:102]: Maybe better to use this style:
```python
expected_output = ("first "
                   "second")
```
then you can avoid this split/strip/join hack

**Review by martinpitt — CHANGES_REQUESTED** [review]
Very nice!

**Review by martinpitt — APPROVED** [review]
Very cool, thanks!

**Pull request merged**